src/cls/IPM/Repo/Oras/PackageService.cls

```diff
@@ -126,11 +126,73 @@ Method ListModulesFromTagString(
             set tag = tag _ $$$OrasTagPlatformSeparator _ platformVersion
         }
 
-        #; get metadata from annotations
+        // get metadata from annotations
         set metadata = ..GetPackageMetadata(..Location, name, tag, "", client)
         set artifactMetadata = ##class(%IPM.Repo.Oras.ArtifactMetadata).%New()
         $$$ThrowOnError(artifactMetadata.%JSONImport(metadata))
 
+        // Filter by module name if requested - check both Name and ExternalName
+        if (searchCriteria.Name '= "") {
+            // Extract the package name without namespace prefix for comparison
+            // Package name from ORAS may be "namespace/modulename" or just "modulename"
+            set packageNameWithoutNamespace = name
+            if (name [ "/") {
+                #; Extract everything after the last "/"
+                for i=$length(name):-1:1 {
+                    if ($extract(name, i) = "/") {
+                        set packageNameWithoutNamespace = $extract(name, i+1, *)
+                        quit
+                    }
+                }
+            }
+
+            // First check if package name already matches
+            if ($$$lcase(packageNameWithoutNamespace) = $$$lcase(searchCriteria.Name)) {
+                set packageNameMatches = 1
+            } else {
+                set packageNameMatches = 0
+            }
+
+            if 'packageNameMatches {
+                // Package name doesn't match - fetch module.xml to check namespaced Name and ExternalName
+                // Note: pass empty namespace since 'name' already includes the full path
+                set moduleXMLText = ..GetModuleXML(..Location, name _ ":" _ tag, "", ..Username, ..Password, ..Token, ..TokenAuthMethod)
+
+                if (moduleXMLText '= "") {
+                    // Parse module.xml
+                    try {
+                        set moduleObj = ##class(%IPM.Utils.Module).GetModuleObjectFromString(moduleXMLText, .found)
+                    } catch ex {
+                        // if a single module.xml fails to parse, move onto the next one instead of failing completely
+                        continue
+                    }
+                    if 'found {
+                        continue
+                    }
+
+                    // Check if search name matches either Name or ExternalName (case-insensitive)
+                    set matchesName = 0
+                    set matchesExternalName = 0
+
+                    if ($$$lcase(moduleObj.Name) = $$$lcase(searchCriteria.Name)) {
```

Comment:
Same with `matchesName` and `matchesExternalName` as above. Can make these single declarations.